src/impl/move_gen.c

```diff
@@ -894,19 +829,10 @@ void generate_moves(Game *game, move_record_t move_record_type,
     gen->current_row_index = anchor_get_row(anchor_list, i);
     gen->last_anchor_col = anchor_get_last_anchor_col(anchor_list, i);
     gen->dir = anchor_get_dir(anchor_list, i);
-    board_set_transposed(gen->board, anchor_get_transposed(anchor_list, i));
-    load_row_letter_cache(gen, gen->current_row_index);
-    load_bonus_square_cache(gen, gen->current_row_index);
-    load_is_cross_word_cache(gen, gen->current_row_index);
-    load_cross_score_cache(
-        gen, gen->current_row_index, !board_is_dir_vertical(gen->dir),
-        board_get_cross_set_index(gen->kwgs_are_shared, gen->player_index));
-    load_cross_set_cache(
-        gen, gen->current_row_index, !board_is_dir_vertical(gen->dir),
-        board_get_cross_set_index(gen->kwgs_are_shared, gen->player_index));
-    load_hz_cross_set_cache(
-        gen, gen->current_row_index,
-        board_get_cross_set_index(gen->kwgs_are_shared, gen->player_index));
+    bool anchor_transposed = anchor_get_transposed(anchor_list, i);
+    gen->row_cache =
+        board_get_const_grid(gen->board, anchor_transposed)->squares +
+        gen->current_row_index * BOARD_DIM * sizeof(Square);
```

Comment:
i know this is wip and maybe you've already fixed this locally but i think you don't want the * sizeof(Square) here. square_ptr + k already adds k*sizeof(Square) to the address